hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1202,6 +1218,7 @@ public void closeContainer(Container container)
       }
       container.close();
       ContainerLogger.logClosed(container.getContainerData());
+      createContainerMerkleTree(container);
```

Comment:
Similar to Ritesh's existing comment, this method will exit if the container has moved from open to unhealthy and won't generate the file. We need tests that we always get a tree when the container leaves the open state. This might be easier as unit tests, unless we can inject a failure into the existing integration test.